modules/querying/pages/query-optimizer/enable-cost-optimizer.adoc

```diff
@@ -0,0 +1,96 @@
+= Enabling Cost-Based Optimizer
+:sectnums:
+:description: Steps to enable the cost-based optimizer.
+
+This page walks you through the steps to enable the cost-based optimizer.
+
+== Before you begin
+* You have created a query that contains a xref:select-statement/index.adoc[`SELECT` statement] that uses a xref:select-statement/index.adoc#_path_pattern[path pattern] in its `FROM` clause with one or more hops.
+* You have the privilege to install queries on the graph where you are operating.
+
+== Procedure
+The following is the procedure to enable cost-based optimizer when installing a query.
+
+=== Identify referenced types and attributes
+You need to identify the vertex and edge types for which you need to compute cardinality data, and attributes of those types for which you need to compute histogram data.
+
+Vertex and edge types:
+
+* Any vertex type referenced in the path pattern of a `FROM` clause.
+* Any _refined edge type_.
+A _refined edge type_ means an edge type with a specified pair of `from` and `to` vertex types.
+
+Attributes:
+
+* Any attribute referenced in a xref:select-statement/index.adoc#_where[`WHERE` clause] in a `SELECT` statement in the query.
+
+For example, if you want to optimize the following query:
+
+[.wrap,gsql]
+----
+CREATE OR REPLACE QUERY example3() FOR GRAPH ldbc_snb SYNTAX V2 {
+
+ vSet = SELECT p
+        FROM Tag:t - (<HAS_TAG) - Comment:m - (<LIKES:e) - Person:p
+        WHERE p.gender == "female" AND t.id != 100;
+
+}
+----
+
+We would need the following statistics:
+
+* Cardinality
+** Vertex types
+*** `Tag`
+*** `Comment`
+*** `Person`
+** Refined edge types
+*** `HAS_TAG`: from `Comment` to `Tag`
+*** `LIKES`: from `Person` to `Comment`
+* Histogram
+** `gender` attribute of type `Person`
+** `id` attribute of type `Tag`
+
+
+=== Compute cardinality data
+To compute cardinality data, use the `POST :14240/gsqlserver/gsql/stats/card` endpoint.
+
+For example, to compute the cardinality data of the vertex `Person` in graph `ldbc_snb`, make the following request:
+
+[source.wrap,console]
+----
+curl -s --user tigergraph:tigergraph -X POST "localhost:14240/gsqlserver/gsql/stats/card?graph=ldbc_snb&vertex=Person"
+----
+
+For another example, to compute the cardinality data of the refined edge type `LIKES` from a `Person` vertex to a `Comment` vertex, make the following request:
+
+[.wrap,console]
+----
+curl -s --user tigergraph:tigergraph -X POST "localhost:14240/gsqlserver/gsql/stats/card?graph=ldbc_snb&edge=LIKES&from=Person&to=Comment"
+----
+
+You need to compute the cardinality data for every type you identified in the previous step, each with its own request.
```

Comment:
Instead of saying `You need to`, I would say something like
```
In order to get the best results, it is recommended to compute cardinality statistics for every vertex and edge type referenced in the query.
```
Also I don't know that we need to call out that each computation requires a separate request here. I think it should be clear given the API parameters.

Reply:
I will add "for best results" to the end of the sentence. We want to be concise when we can; 

As for emphasizing that each computation requires its own request, I felt like it's good to have there because I was not sure about that from reading the parameters and had to ask you. I feel like if I was a user and had a very long query with dozens of types, I would probably wonder if I need to make that many requests - it can sound a little bad, but it's worse if they have to wonder and then find out.